sdks/python/container/license_scripts/dep_urls_py.yaml

```diff
@@ -0,0 +1,118 @@
+  #   license: url_to_license or "skip"
+  #     # skip: don't add license for the dependency
+  #     # url_to_license: a url to source code (https://...) or a local file (file://...)
+  #   notice: it's optional, url_to_notice
+  apache-beam:
+    license: "skip" # don't include self dependency.
+  # TODO(BEAM-8371): License for this dependency cannot be automatically retrieved on Python 2.
+  appdirs:
+    license: "https://raw.githubusercontent.com/ActiveState/appdirs/master/LICENSE.txt"
+  avro:
+    license: "https://raw.githubusercontent.com/apache/avro/master/LICENSE.txt"
+    notice: "https://raw.githubusercontent.com/apache/avro/master/NOTICE.txt"
+  avro-python3:
+    license: "https://raw.githubusercontent.com/apache/avro/master/LICENSE.txt"
+    notice: "https://raw.githubusercontent.com/apache/avro/master/NOTICE.txt"
+  backports.weakref:
+    license: "https://raw.githubusercontent.com/PiDelport/backports.weakref/master/LICENSE"
+  chardet:
+    license: "https://raw.githubusercontent.com/chardet/chardet/master/LICENSE"
+  certifi:
+    license: "https://raw.githubusercontent.com/certifi/python-certifi/master/LICENSE"
+  cython:
+    license: "https://raw.githubusercontent.com/cython/cython/master/LICENSE.txt"
+  enum34:
+    license: "https://bitbucket.org/stoneleaf/enum34/raw/c208549a93b71b948ff7bbdfd29dce8f85527916/enum/LICENSE"
+  fastavro:
+    license: "https://raw.githubusercontent.com/fastavro/fastavro/master/LICENSE"
+    notice: "https://raw.githubusercontent.com/fastavro/fastavro/master/NOTICE.txt"
+  funcsigs:
+    license: "https://raw.githubusercontent.com/aliles/funcsigs/master/LICENSE"
+  futures:
+    license: "https://raw.githubusercontent.com/agronholm/pythonfutures/master/LICENSE"
+  google-apitools:
+    license: "https://raw.githubusercontent.com/google/apitools/master/LICENSE"
+  googledatastore:
+    license: "https://raw.githubusercontent.com/GoogleCloudPlatform/google-cloud-datastore/master/LICENSE"
+  grpcio:
+    license: "https://raw.githubusercontent.com/grpc/grpc/master/LICENSE"
+    notice: "https://raw.githubusercontent.com/grpc/grpc/master/NOTICE.txt"
+  grpcio-gcp:
+    license: "https://raw.githubusercontent.com/GoogleCloudPlatform/grpc-gcp-python/master/LICENSE"
+  guppy:
+    license: "https://raw.githubusercontent.com/joshwcomeau/guppy/master/LICENSE.md"
+  guppy3:
+    license: "https://raw.githubusercontent.com/zhuyifei1999/guppy3/master/LICENSE"
+  h5py:
+    license: "https://raw.githubusercontent.com/h5py/h5py/master/LICENSE"
+  hdfs:
+    license: "https://raw.githubusercontent.com/mtth/hdfs/master/LICENSE"
+  httplib2:
+    license: "https://raw.githubusercontent.com/httplib2/httplib2/master/LICENSE"
+  mock:
+    license: "https://raw.githubusercontent.com/testing-cabal/mock/master/LICENSE.txt"
+  monotonic:
+    license: "https://raw.githubusercontent.com/atdt/monotonic/master/LICENSE"
+  nose:
+    license: "https://raw.githubusercontent.com/nose-devs/nose/master/lgpl.txt"
```

Comment:
FWIW, I think we can remove 'nose' from our containers. I don't see a reason why we include it in https://github.com/apache/beam/blob/296e94b6112e833e6d467d5f9e8dfc83789aaa09/sdks/python/container/base_image_requirements.txt#L47. It's a test-only dependency.

Reply:
I agree with it. I will create a new PR to test behaviors when we remove dependencies.